parsec/remote_dep_mpi.c

```diff
@@ -1676,12 +1682,14 @@ remote_dep_mpi_put_start(parsec_execution_stream_t* es,
                                    -1,
                                    &source_memory_handle, &source_memory_handle_size);
         } else {
-            /* TODO: Implement converter to pack and unpack */
-            int dtt_size;
-            parsec_type_size(dtt, &dtt_size);
+            /* TODO: Implement converter to pack and unpack
+             * register the whole region including the holes because we don't support sparse
+             * registration. */
+            ptrdiff_t extent, lb;
+            parsec_type_extent(dtt, &lb, &extent); (void)lb;
             parsec_ce.mem_register(dataptr, PARSEC_MEM_TYPE_CONTIGUOUS,
-                                   -1, NULL, // TODO JS: this interface is so broken, fix it!
-                                   dtt_size, // TODO JS: what about nbdtt? Is it ok to ignore it?!
+                                   -1, parsec_datatype_uint8_t,
+                                   nbdtt * extent,
```

Comment:
I have implemented your solution @bosilca, but I can't decide if that should be `nbdtt * (extent-lb)`

Reply:
Definitely not `nbdtt * (extent-lb)`. Visualize each data in term of the true lb and true ub, which are the first and last byte used. If there are gaps in the middle, they are part of the true extent. Technically the span of memory you want to cover start after the initial gap (aka the first byte if the first instance of an element), and should go until the last byte (the true ub) of the last instance of the element. But this would require to move the initial pointer by the first gap (true lb of the first element), and this complexify the pointer manipulation. `nbdtt * extent` would cover more memory for all well-formed datatypes.